index.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+const symbols = require('fastify/lib/symbols')
+const Middie = require('./engine')
+const kMiddlewares = Symbol('fastify-middie-middlewares')
+const kMiddie = Symbol('fastify-middie-instance')
+
+function middiePlugin (fastify, options, next) {
+  // TODO: we should use decorate, but `use` is already
+  // a public API of Fastify. In Fastify v3 it will be deprecated,
+  // so we will able to use `decorate`
+  fastify.use = use
+  fastify[kMiddlewares] = []
+  fastify[kMiddie] = Middie(onMiddieEnd)
+
+  fastify
+    .addHook('onRequest', enhanceRequest)
+    .addHook('onRequest', runMiddie)
+    .addHook('onRegister', onRegister)
+
+  function use (path, fn) {
```

Comment:
In examples are using multiples paths: `['css/*', 'js/*']`. I think that here is not covering it. Right?

Reply:
It should be covered in the Middie engine tests.